### ISSUE TITLE
Glyph smaller than character

### ISSUE THREAD

**Issue by kiuKisas** [ryanoasis/nerd-fonts#127]
#### Requirements

If you are **not** sure whether or not what you are reporting is actually an issue **or** you have a general question please message in the [Gitter Chat Room](https://gitter.im/ryanoasis/nerd-fonts)

By posting an issue you acknowledge the following:

- [x] A brief but descriptive _title_ of your issue
- [x] I have searched the [issues](https://github.com/ryanoasis/nerd-fonts/issues) for my issue and found nothing related and/or helpful
- [x] I have read or at least glanced at the [FAQ](https://github.com/ryanoasis/nerd-fonts#faq--troubleshooting)
- [x] I have read or at least glanced at the [Wiki](https://github.com/ryanoasisnerd-fonts/wiki)

These items you must provide answers to. Make sure to add **all the information needed to understand the issue** so that someone can help. If the info is missing we'll add the 'Needs more information' label and _may_ choose to close the issue until there is enough information.

**^ ----- Please remove all of the above _and_ this line before posting your issue ----- ^**

#### Subject of the issue

Glyph looks smaller than characters in `neovim`, with `vim-devicons`

#### Your Setup

- Knack Nerd Font (download from the github)
- termite with zsh
-  Arch Linux - kernel 4.8.13-1-ARCH 

#### Optional

![2017-01-26-135412_958x1050_scrot](https://cloud.githubusercontent.com/assets/12698940/22325271/4453d0d4-e3d3-11e6-97c9-90ae0f98e19a.png)

**Comment by jayywolff** [comment]
I can also confirm the small icons, and I have the same setup but except I'm on the Fura Mono Font
x86_64 Linux 4.8.13-1-ARCH

with the pkg version  nerd-fonts-complete 1.0.0-1
I have this issue with every font I've tried so far 
including (inconsolata, fura mono, meslo, terminess, dejavu sans, knack) 
This is not just a terminal specific issue Ive noticied this in termite,  i3bar , & gvim

![2017-01-27-000148_1920x1080_scrot](https://cloud.githubusercontent.com/assets/16052485/22361040/caf94664-e423-11e6-8298-663860364d13.png)

**Comment by ryanoasis** [comment]
Ouch. Thank you both for the details. I will try to take a look this weekend if I get the chance

**Comment by ryanoasis** [comment]
Can either/both of you tell me the exact font using? Most importantly if it ends in `Complete Mono` ?

**Comment by jayywolff** [comment]
I'm using FuraMonoForPowerline Nerd Font Regular 11 in the terminal.

**Comment by jayywolff** [comment]
This very likely may not have been an upstream bug. I noticed all of the affected users are running the AUR version of nerd-fonts. There was an update to the AUR package today where the package maintainer noted he removed he removed the "single-width-glyphs" variant fonts from the package so the system would use the "non-single-width-glyphs" fonts. I can confirm the upgrade of the AUR package nerd-fonts-complete 1.0.0-2 -> 1.0.0-3,  fixed this bug for me.

**Comment by ryanoasis** [comment]
@jayywolff Thanks for the update. Good observation on those affected.

I think the root cause/fix is to make sure the 'mono' versions internal name have 'mono' somewhere so they don't conflict with 'non-mono'. Right now if you install both it is seen as one, as far as I know.

![selection_010](https://cloud.githubusercontent.com/assets/8083459/25764229/dd882a5a-31b4-11e7-9839-ccd09a19fad0.png)

**Comment by ryanoasis** [comment]
@kiuKisas can let us know if @jayywolff suggestion works for you?

Thanks :smile:

**Comment by RuRo** [comment]
I seem to be having the same issue. I am using the `Knack Regular Nerd Font Complete Mono` font. I installed it by downloading the `.ttf` files from the `1.0.0` branch. Arch (Manjaro): Linux 4.9.24-1.

Most characters are so small, that they are pretty much not usable as glyphs and indicators. Some of the characters are as big as they should be. For example the powerline symbols are normal sized, while the various git logos are very small.

Compare sizes:
`echo "\nSmall :\uf300\uf460\ue70a\ue258\nSo-so :\uf000\uf072\nNormal:\ue285\uf245\n"`
![screenshot_20170507_141114](https://cloud.githubusercontent.com/assets/3747318/25780354/14ce17d8-332f-11e7-9e88-ba9dc044516f.png)

From Character Map App:
![screenshot_20170507_141449](https://cloud.githubusercontent.com/assets/3747318/25780374/8e8b66ca-332f-11e7-9fed-8564be772f1b.png)

I don't think this is font dependant, or at least not just `Knacks` faulta, because previously I had this same issue with `Sauce Code Pro`. I even think the same characters were small.

**Comment by ryanoasis** [comment]
@RuRo thanks for the details. can you please try the `Knack Regular Nerd Font Complete` (non-mono) variant?

the `Mono` variants it is expected that the glyphs are a bit on the small size (to main scale in a fixed-width glyph)

**Comment by RuRo** [comment]
@ryanoasis I will try it, when I get home, however that doesn't make much sense to me. The characters, that are too small are even smaller than English letters as well as some other Mono-spaced characters. There is enough space in a character to increase the size of the icon. Or at least it looks like that to me.

If it works with non-mono font, that wouldn't really be a fix, since these fonts are mostly used for terminals and you can't use mono-spaced fonts in terminals. If increasing the size of icons is not possible in mono-spaced fonts, could you make 2 character wide variants for the small icons. I.e. the left part character + the right part? Thank you.

**Comment by ryanoasis** [comment]
@RuRo Okay sure let me know how it goes. :smile:

As for the size, I think it just _looks_ that way as you say. It is a bit deceiving until you highlight or change the background color that it becomes more obvious:

![selection_001](https://cloud.githubusercontent.com/assets/8083459/25879303/eeda9296-34fe-11e7-9130-544e3bb7cf98.png)

The first 2 glyphs look like they _could_ be a bit larger but not much without the ratio changing (stretching)

**Comment by RuRo** [comment]
Well. I tried the regular size font. As expected, the characters are bigger, but it is completely broken in terminal. The git-in-a-circle character is still small btw, but whatever.
![spectacle m20477](https://cloud.githubusercontent.com/assets/3747318/26028130/ad275772-3822-11e7-9837-0157a4124a9a.png)

Is there really no way to have the cake and eat it? Maybe something hacky, like splitting the smaller characters into 2 monospaced parts.

Or maybe allow non-monospace characters, but have their width be a multiple of a single character, so then you could do something like this in terminal:
```
<double width icon><zero width char> is the same length as
<normal character><normal character> 
```

**Comment by ryanoasis** [comment]
@RuRo First thanks for your thoughts and ideas. I can sense some frustration you may have had, if so sorry about that :worried: 

> Is there really no way to have the cake and eat it? Maybe something hacky, like splitting the smaller characters into 2 monospaced parts.

Hmm yeah this _is_ an interesting idea. I actually thought about doing something like this too for issues we've had with ambiguous glyph width handling in Vim but I am not sure how feasible it is and managing this I can see being a nightmare :worried: 

For the "double width" sorry if there was any confusion. If you have any ideas for making things more clear in the readme please let me know.

So I guess you realized that the way to go is to use the 'double width' variant (depending on terminal support).

> P.S. Konsole and Yakuake don't seem to support double width characters. Can somebody reccomend a drop down terminal that does?

Guake Terminal seems to support non-mono fonts.

**Comment by RuRo** [comment]
Hm-hmmm. I tried the double-width variant with Guake. The big characters work well, but now the powerline arrows are slightly misaligned the '<' arrows have a small space between them and the next block, while the '>' arrows are vertically a little smaller than the whole 'line'. Here's a screenshot:
![screenshot_20170521_014818](https://cloud.githubusercontent.com/assets/3747318/26279846/aa1a7e1a-3dc7-11e7-9457-da8353e8c56b.png)

Actually, maybe the git symbol is a bit too small. But it works unexpectedly good overall.
![screenshot_20170521_015051](https://cloud.githubusercontent.com/assets/3747318/26279852/f7c617dc-3dc7-11e7-9f12-384b1ea3b23b.png)

**Comment by ryanoasis** [comment]
@RuRo Thanks for the details.

Can you tell me which font and font size you are using by the way?

Hopefully the alignment issues will at least be _mostly_ solved in v1.1.0, here is how it is showing for me:

![selection_006](https://cloud.githubusercontent.com/assets/8083459/26280582/b84d9b22-3da7-11e7-8b55-d469ff14e86d.png)

As for the GitHub logo you can try to use the 'other' set:

![selection_007](https://cloud.githubusercontent.com/assets/8083459/26280586/e1e87d9e-3da7-11e7-9d05-ee3cf19d3f33.png)

**Comment by ryanoasis** [comment]
Fonts just uploaded... probably won't improve anything here but if anyone would like to test them out that'd be great :smile:

**Comment by shibumi** [comment]
Hello,
I have installed the following Fonts:
```
https://raw.githubusercontent.com/ryanoasis/nerd-fonts/v1.0.0/patched-fonts/FiraCode/Bold/complete/Fura Code Bold Nerd Font Complete Mono.otf
https://raw.githubusercontent.com/ryanoasis/nerd-fonts/v1.0.0/patched-fonts/FiraCode/Light/complete/Fura Code Light Nerd Font Complete Mono.otf
https://raw.githubusercontent.com/ryanoasis/nerd-fonts/v1.0.0/patched-fonts/FiraCode/Medium/complete/Fura Code Medium Nerd Font Complete Mono.otf
https://raw.githubusercontent.com/ryanoasis/nerd-fonts/v1.0.0/patched-fonts/FiraCode/Regular/complete/Fura Code Regular Nerd Font Complete Mono.otf
https://raw.githubusercontent.com/ryanoasis/nerd-fonts/v1.0.0/patched-fonts/FiraCode/Retina/complete/Fura Code Retina Nerd Font Complete Mono.otf
https://raw.githubusercontent.com/ryanoasis/nerd-fonts/v1.0.0/patched-fonts/FiraMono/Bold/complete/Fura Mono Bold for Powerline Nerd Font Complete Mono.otf
https://raw.githubusercontent.com/ryanoasis/nerd-fonts/v1.0.0/patched-fonts/FiraMono/Medium/complete/Fura Mono Medium for Powerline Nerd Font Complete Mono.otf
https://raw.githubusercontent.com/ryanoasis/nerd-fonts/v1.0.0/patched-fonts/FiraMono/Regular/complete/Fura Mono Regular for Powerline Nerd Font Complete Mono.otf
https://raw.githubusercontent.com/ryanoasis/nerd-fonts/v1.0.0/patched-fonts/Inconsolata/complete/Inconsolata for Powerline Nerd Font Complete Mono.otf
```

However when I want to choose a font in `qterminal` I see only these fonts:
```
![](https://paste.archlinux.de/PQd/)

I can confirm the  small glyphs, too:
![](https://paste.archlinux.de/dIuI/)

But now starts the funny part. In my i3blocks statusbar I have the right size of the glyphs:

![](https://paste.archlinux.de/eqn2Hx/)

But that seems to have something todo with: https://github.com/gabrielelana/awesome-terminal-fonts
I have installed them as well and when I remove them I have small icons just like in my terminal..

**Comment by towo** [comment]
I have the funny situation of having both small and good-sized glyphs... on different machines.
The zsh config is shared via git (not public yet due to ... history issues), so the configuration is the same, but one machine has okay-sized icons, the other has the small glyphs; both are using `SauceCodePro Nerd Font Mono Regular 9`.

Font selection on the machine with *small* glyphs:
```
$ pacaur -Qe G font 
gnome-font-viewer 3.26.0-1
nerd-fonts-complete 1.2.0-3
noto-fonts 20171025-2
noto-fonts-emoji 20180102-2
powerline-fonts 2.6-1
terminus-font 4.46-1
ttf-win7-fonts 7.1-9
```

On the machine with big glyphs:
```
$ pacaur -Qe G font 
gnome-font-viewer 3.26.0-1
nerd-fonts-complete 1.2.0-3
noto-fonts-emoji 20180102-2
powerline-fonts 2.6-1
```

I'll try reducing the font list to the one of the latter. This'll take a bit due to logout/docking station suspending hijinx.

**Comment by ryanoasis** [comment]
Thanks @shibumi sorry for the very late reply. I'll try to take a look at Inconsolata

**Comment by benbuch** [comment]
Its the same issue on SauceCode Pro Nerd Font Mono
![2018-05-29-134338_1920x1080_scrot](https://user-images.githubusercontent.com/6019639/40654249-675a91aa-6346-11e8-8e3b-a426aa748878.png)

**Comment by benbuch** [comment]
This is also a issue with SauceCodePro Nerd Font Mono. i have tried both the nerd font complete from aur and also download from nerdfont's website.
![2018-05-29-143958_1920x1080_scrot](https://user-images.githubusercontent.com/6019639/40656806-402e5e4c-634e-11e8-8ab5-9bc06c28d6cc.png)

**Comment by yoyoys** [comment]
same problem on DejaVu Sans Mono but only in Ubuntu (18)
terminal: hyper 2.0
screenshot on ubuntu:
![image](https://user-images.githubusercontent.com/22124891/41192961-e8f79f64-6c38-11e8-8402-f11815d731fd.png)

my mac works fine:
![image](https://user-images.githubusercontent.com/22124891/41192959-d09ad6b6-6c38-11e8-8f30-b700b03133f6.png)

**Comment by FelipeMarcelino** [comment]
Same here. Fura Code and Iosevka in arch with i3

**Comment by jayywolff** [comment]
Hey guys, I haven't experienced this issue for while now. Last year it would fix and break between some updates, but the AUR package has been stable lately.

Currently using the AUR package nerd-fonts-complete 2.0.0-5
Still running Arch, i3wm, termite as my terminal emulator.
Currently using Iosevka Nerd Font

**Comment by myrage2000** [comment]
I have exactly the same config (ArchLinux, i3wm, nerd-fonts-complete) but
with urxvt. Could you try with it please ?

Le jeu. 13 sept. 2018 à 21:44, Johnathan Wolff <notifications@github.com> a
écrit :

**Comment by jayywolff** [comment]
@myrage2000 what do you have for your "URxvt.font" setting in .Xresources? I've never been able to get icons to work at all in urxvt

**Comment by myrage2000** [comment]
If i'm right I have something like : "xft:[insert random nerd-font MONO
here]:size=8"
And glyphs are 2x smaller than letters.
Maybe urxvt does not support it so ? This is weird I have never heard that

Le jeu. 13 sept. 2018 à 22:10, Johnathan Wolff <notifications@github.com> a
écrit :

**Comment by ryanoasis** [comment]
@myrage2000 can you provide a screenshot? 

This is probably to be expected in URxvt, see: https://github.com/ryanoasis/nerd-fonts/wiki/Terminal-Emulators#urxvt

**Comment by jayywolff** [comment]
@myrage2000  

with the setting "URxvt.font: xft:Iosevka Nerd Font Mono:size=16" I am able to see the glyphs but they are smaller than normal characters and several glyphs are missing for some reason.

**Comment by ykyuen** [comment]
i installed two nerd-fonts on my archlinux laptop with i3-wm
1. `InconsolataGo`
2. `Noto`

in i3 config, when i set the font to `font pango:Noto Sans 20`, small glyph was found.

but the size returns to normal when i set to `font pango:InconsolataGo Nerd Font 20`

Maybe u could also try `InconsolataGo Nerd Font`?

**Comment by M3rsy** [comment]
> I seem to be having the same issue. I am using the `Knack Regular Nerd Font Complete Mono` font. I installed it by downloading the `.ttf` files from the `1.0.0` branch. Arch (Manjaro): Linux 4.9.24-1.
> 
> Most characters are so small, that they are pretty much not usable as glyphs and indicators. Some of the characters are as big as they should be. For example the powerline symbols are normal sized, while the various git logos are very small.
> 
> Compare sizes:
> `echo "\nSmall :\uf300\uf460\ue70a\ue258\nSo-so :\uf000\uf072\nNormal:\ue285\uf245\n"`
> ![screenshot_20170507_141114](https://cloud.githubusercontent.com/assets/3747318/25780354/14ce17d8-332f-11e7-9e88-ba9dc044516f.png)
> 
> From Character Map App:
> ![screenshot_20170507_141449](https://cloud.githubusercontent.com/assets/3747318/25780374/8e8b66ca-332f-11e7-9fed-8564be772f1b.png)
> 
> I don't think this is font dependent, or at least not just `Knacks` fault, because previously I had this same issue with `Sauce Code Pro`. I even think the same characters were small.

What Programm are you using to see the icons with code unicode?

**Comment by kajzersoze** [comment]
Irritating problem, some fonts have small glyphs some don't.

Linux, gentoo
Terminal : st (had the same problem in urxvt)

**Comment by iamkarlson** [comment]
The same problem for windows. 
![image](https://user-images.githubusercontent.com/4534374/50898729-4fa8b700-1410-11e9-9dc6-29070769e436.png)

**Comment by jayywolff** [comment]
I've just been dealing with the tiny icons for almost 2 years now lol. At some point  in 2017 there was an update that fixed it and then the following update broke it again. Super annoying.

**Comment by Ratione** [comment]
Okay, has there been any serious attempt to solve this, by someone who knows literally anything about fonts? Because I don't.

I really like the Nerd-Fonts, and I don't want to go find something else. If more data is needed, I would be happy to help, but I don't even know how to start solving this problem.

**Comment by caesarsol** [comment]
Hi, sorry to jump in, I might have a similar problem: 
I have just downloaded and installed the Ubuntu Mono Nerd Font, using it in iTerm on Mac OS X, and the glyphs appear pretty small (especially the Directory icon).

Here is a screenshot:
![image](https://user-images.githubusercontent.com/1799710/53162334-19905280-35cc-11e9-96ba-ab63d5dd4c6a.png)


I arrived here through [this project (LSD)](https://github.com/Peltoche/lsd), and its homepage shows a screenshot with much better-sized icons:
![image](https://user-images.githubusercontent.com/1799710/53162398-3b89d500-35cc-11e9-9f84-831a1faa00e6.png)


Do you think it's the same issue or am I doing something wrong?
Thanks for the help!

**Comment by deimosfr** [comment]
Same for me, on Arch with Inconsolata

**Comment by MarinPostma** [comment]
Any update on this issue? weird thing is that it works perfectly on my Manjaro install but not on my Arch...

**Comment by iamkarlson** [comment]
Hi guys, I'm not sure that it works for all but simply removing the font and installing it again solved my problem in windows env. Just make sure that you used the right file for the right OS

**Comment by jayywolff** [comment]
> Any update on this issue? weird thing is that it works perfectly on my Manjaro install but not on my Arch...

I have it working for me on Arch on multiple machines, and Ive gotten it to work on Ubuntu as well, but it doesn't just work out of the box, I've never got it working for any font in urxvt, and I cant get it working with any mono-spaced fonts. I've gotten used to not using mono fonts, its worth it for me, would rather have awesome icons for all the things than equal width characters.

I feel like its been different to get it working on every machine but I haven't done it enough times to be sure. I think the trick is making sure you have the correct fallback font(s) installed/configured.

Im still using the  Iosevka Nerd Font Complete. In my setup I must also have the stock Iosevka font installed from Arch's repo or else it doesn't work properly

**Comment by sztomi** [comment]
I also have a problem with the glyph sizes. I've found the following:

  * gnome-terminal doesn't let me select the nerd fonts at all. There was another thread somewhere explaining that some value is not set in to fonts which causes this (I can't find that now, unfortunately)
  * in Kitty, I get the small glyphs issue
  * in Konsole, I initially can't select them, but when I check "show all fonts", it lets me select the nerdfonts and they render correctly (i.e. no small glyphs!)

Kitty:

![image](https://user-images.githubusercontent.com/240594/58761555-e273fa80-8545-11e9-9b6c-4c8347ee261c.png)

Konsole:

![image](https://user-images.githubusercontent.com/240594/58761540-cc663a00-8545-11e9-9cfd-54f7570e1e8d.png)

I think this missing value is related to the incorrect rendering.

**Comment by sstark** [comment]
>     * gnome-terminal doesn't let me select the nerd fonts at all. There was another thread somewhere explaining that some value is not set in to fonts which causes this (I can't find that now, unfortunately)

gnome-terminal only lists fonts that are marked monospaced, which seems is not the case for nerd fonts. You can trick it to use them anyway by unchecking "user font" in gnome-terminal and choose the font you want as the system-wide monospace font in gnome-tweaks.

More on-topic: With lsd it is simple to show the issue by comparing the output of `lsd --icon-theme fancy` with `lsd --icon-theme unicode`. The latter for instance shows correctly sized folder icons, the fancy version (which uses all the extra glyphs) doesn't.

**Comment by nsklaus** [comment]
i have the issue too, i installed the already patched inconsolata version, then i ran `fc-cache -f -v`.
this is on ubuntu 19.04, i3wm and st terminal. the letters are normal but the symbols are very small.
less than half the height of the letter.

**Comment by abstracted** [comment]
@sztomi If you're using kitty, you will have to set the symbol map to the non-monospaced version of your nerdfont. 

It gets set to the monospaced version automatically due to the fact kitty will automatically use monospace fonts by default.

Ex:
`symbol_map U+E000-U+FFFF [NERD_FONT_NAME]`

you can find your installed fonts with

`fc-list | cut -d: -f2 | sort -h | uniq | grep -i nerd`

**Comment by waltereikrem** [comment]
###Under  you can see that my icons are still super tiny. I have tried about 6 different nerdfonts now. some from github, some from the AUR, some mone, some complete. and still, like the image icon is almost as small as a "*".  while others look super chunky.

![my issue](https://i.imgur.com/ZzyH3YA.png)

###Font settings atm
![my issue](https://i.imgur.com/XrydUJk.png)

as you can see. im using konsole, witht the font Source Code Pro. 


Any help? i see this is randomly an issue for people on this thread going almost 3 years back?

**Comment by Alexey104** [comment]
I have the same issue with the most of fonts, but 'nerd-fonts-ubuntu-mono' work fine for me in any terminal on Arch:
![screenshot](https://user-images.githubusercontent.com/38672094/74370940-01bb6b00-4de9-11ea-85c3-0aa54266e5f7.png)

**Comment by paperbenni** [comment]
I have the problem in dwm and its a lot worse there

**Comment by atreyasha** [comment]
With the recent upgrade in the Arch AUR of `nerd-fonts-complete` to version 2.1.0-1, I experienced the issue of glyphs becoming very small compared to characters. I am using`i3-gaps` and my font configuration is:

```
font pango: 3270 Nerd Font 8.5
```

I solved the problem by uninstalling the `nerd-fonts-complete` AUR package and only installing the `3270` complete font package directly from this repository. Perhaps there was some conflict with the large `nerd-fonts-complete` package. Hope this information could help someone else facing a similar problem.

**Comment by jaisharx** [comment]
_**_Issue becomes apparent when compared with how the default vscode's terminal shows the exact same glyphs, also I'm using the cascadia code in both the terminal._**_ 


![Annotation 2020-05-22 075739](https://user-images.githubusercontent.com/55892116/82625229-3e75da00-9c02-11ea-80d4-d4d14826087e.png)

![Annotation 2020-05-22 080227](https://user-images.githubusercontent.com/55892116/82625485-dd023b00-9c02-11ea-825a-f791f9a3de5d.png)


We can see clearly how the arrow, tilde or the X is shown in both the terminal, there's a huge difference. The arrow and the X almost makes it apparent.

**Comment by scarolan** [comment]
I have this issue on WSL but only in Windows Terminal. Visual Studio Code's built in terminal renders the glyphs at the right size.

VSC Terminal:
![image](https://user-images.githubusercontent.com/403332/87855039-5c4a7d00-c8db-11ea-84c4-98b8b8c719b6.png)

Windows Terminal:
![image](https://user-images.githubusercontent.com/403332/87855049-679da880-c8db-11ea-8a99-0aac8eca9a47.png)

**Comment by Animeshz** [comment]
+1 fonts with `Mono` suffix have been suffering with the issue from my experience...

**Comment by colinli** [comment]
> Hi, sorry to jump in, I might have a similar problem:
> I have just downloaded and installed the Ubuntu Mono Nerd Font, using it in iTerm on Mac OS X, and the glyphs appear pretty small (especially the Directory icon).
> 
> Here is a screenshot:
> ![image](https://user-images.githubusercontent.com/1799710/53162334-19905280-35cc-11e9-96ba-ab63d5dd4c6a.png)
> 
> I arrived here through [this project (LSD)](https://github.com/Peltoche/lsd), and its homepage shows a screenshot with much better-sized icons:
> ![image](https://user-images.githubusercontent.com/1799710/53162398-3b89d500-35cc-11e9-9f84-831a1faa00e6.png)
> 
> Do you think it's the same issue or am I doing something wrong?
> Thanks for the help!

Exactly the same issue. 
MacOS big sur
LSD 0.20.1
iTerm2 3.4.9

**Comment by YanDoroshenko** [comment]
Having the same issue, Arch Linux, Alacritty, DejaVu Sans Mono.

Tried installing different packages from AUR, using the install script, nothing seems to work.

**Comment by Finii** [comment]
> @Animeshz @colinli 
> +1 fonts with `Mono` suffix have been suffering with the issue from my experience...

Well... `Mono` means the font is monospaced, i.e. the width of the symbols are limited to the same size as the letters.

![image](https://user-images.githubusercontent.com/16012374/165136036-2e5e2a7f-508e-4495-a9ee-6f9088e47ae1.png)

The upper example is 'too small'? In fact it is not too small, it is just monospaced. The 'document' icon has the same width as the letter `c` i moved below.

On the other hand, the 'ok' behavior is has no monospaced icons, 'document' icon is as wide as a full `a` and half of the `m`.

If you want to have big (wider than letters) icons you can not use `Nerd Font Mono` fonts.

I guess that is the problem in a lot cases here. Unfortunately people do not tell which **exact font files** they use, and as there are too many variants it is hard to tell what they use or misuse.

> @atreyasha @YanDoroshenko 
> I solved the problem by uninstalling the `nerd-fonts-complete` AUR package and only installing the `3270` complete font package directly from this repository.

This is also a problem. The AUR packages are rather far behind and the fonts there have issues that are already solved here.

> @scarolan 
> I have this issue on WSL but only in Windows Terminal.

`Windows Terminal` has a specific handling of the fonts, and it _re-scales_ individual glyphs if it sees fit (i.e. to make the font monospaced (see above)).

**Comment by Finii** [comment]
Arch and how they broke it for everyone else: #634 -> #764, detailed information in https://github.com/ryanoasis/nerd-fonts/issues/731#issuecomment-1002910971 ff

On naming #746 and what `Mono Nerd Font Mono` means ;-)

**Comment by schovi** [comment]
Any solution I can use to solve this by myself?

I have this ["Comic Code" font](https://tosche.net/fonts/comic-code) with added glyphs `font-patcher -c` and the result looks:

<img width="410" alt="CleanShot 2022-05-16 at 17 16 58@2x" src="https://user-images.githubusercontent.com/11453/168626403-a6acb915-7dfc-4609-8489-effca1ab28e1.png">

I tried iTerm2 and Terminal with the same output.

**Comment by schovi** [comment]
> @schovi Your issue looks different. The issue is that the 'line' is higher than the powerline symbols, right?

@Finii probably? (I am really a font noob :) ). Other patched fonts by myself look good, so it is probably related to this specific one?

**Comment by ZhaoZuohong** [comment]
Here is a simple workaround on linux with fontconfig.

Since glyths in monospaced fonts are smaller, we can combine glyphs in sans-serif font and normal characters in monospaced fonts. For example, I personally use *DejaVu Sans Mono* as main font, and take glyphs from *NotoSans Nerd Font*, so I write these settings in `~/.config/fontconfig/fonts.conf`:

```xml
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<fontconfig>
  <dir>~/.fonts</dir>
  <match>
    <test name="family">
      <string>monospace</string>
    </test>
    <edit binding="strong" mode="prepend" name="family">
      <string>DejaVu Sans Mono</string>
      <string>NotoSans Nerd Font</string>
    </edit>
  </match>
</fontconfig>
```

Now those glyphs are bigger. Here is the result:

![image](https://user-images.githubusercontent.com/34163622/203454601-1b713e4a-ffbc-4ba5-a96d-c8f76ce50a1f.png)

**Comment by Finii** [comment]
Here another link with explanantions for people that find this issue
* #969 

Some people reported reinstalling helped. This is because the font family naming is broken and the system takes the wrong font file - sometimes. This will be fixed in `v3.0.0` (major version because the font family names need to change).
See also #1015 as an example.
